File: articles/edsger-dijkstra-and-the-discipline-of-structured-programming/prolog/structured_loop.pl
sum_to(N, Result) :- sum_loop(0, N, 0, Result).

sum_loop(I, N, Acc, Acc) :- I > N.
sum_loop(I, N, Acc, Result) :-
    I =< N,
    Acc1 is Acc + I,
    I1 is I + 1,
    sum_loop(I1, N, Acc1, Result).

% Query example:
% ?- sum_to(5, Result).

File: articles/edsger-dijkstra-and-the-discipline-of-structured-programming/java/StructuredLoop.java
public class StructuredLoop {
    static int sumTo(int n) {
        int acc = 0;
        for (int i = 0; i <= n; i++) {
            acc += i;
        }
        return acc;
    }

    public static void main(String[] args) {
        System.out.printf("sum_to_5=%d%n", sumTo(5));
    }
}
